lib/internal/process/esm_loader.js

```diff
@@ -27,6 +31,27 @@ function initializeImportMetaObject(wrap, meta) {
   if (vmModule === undefined) {
     // This ModuleWrap belongs to the Loader.
     meta.url = wrap.url;
+    let req;
+    defineProperty(meta, 'require', {
+      enumerable: true,
+      configurable: true,
+      get() {
+        if (req !== undefined)
+          return req;
+        const url = new URL(meta.url);
+        const path = getPathFromURL(url);
+        const mod = new Module(path, null);
+        mod.filename = path;
+        req = makeRequireFunction(mod);
+
+        // remove properties that don't affect ESM loading
+        delete req.cache;
+        delete req.extensions;
+        delete req.main;
+        delete req.paths;
```

Comment:
`req.paths` do you mean `req.resolve.paths`?

Reply:
ah, yes

Reply:
Given this confusion - would it be better to keep this to a minimal "only expose the pure function without any additional properties"? E.g. `import.meta.require = actualRequire.bind(null)` to get rid of all potential properties..?